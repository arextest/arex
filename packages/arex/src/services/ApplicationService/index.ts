export * from './createApp';
export * from './deleteApplication';
export * from './getAppInfo';
export * from './getAppList';
export * from './getDependencyList';
export * from './getInterfacesList';
export * from './modifyApp';
