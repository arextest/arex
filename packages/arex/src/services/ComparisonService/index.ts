export * from './batchDeleteIgnoreNode';
export * from './batchInsertIgnoreNode';
export * from './deleteEncryptionNode';
export * from './deleteIgnoreCategory';
export * from './deleteIgnoreNode';
export * from './deleteSortNode';
export * from './deleteTransformNode';
export * from './getTransformMethod';
export * from './insertEncryptionNode';
export * from './insertIgnoreCategory';
export * from './insertIgnoreNode';
export * from './insertSortNode';
export * from './insertTransformNode';
export * from './queryCategoryType';
export * from './queryEncryptionNode';
export * from './queryIgnoreCategory';
export * from './queryIgnoreNode';
export * from './querySortNode';
export * from './queryTransformNode';
export * from './updateEncryptionNode';
export * from './updateSortNode';
export * from './updateTransformNode';
