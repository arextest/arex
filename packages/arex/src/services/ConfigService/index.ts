export * from './agentInstances';
export * from './queryConfigTemplate';
export * from './queryInterfaceResponse';
export * from './queryRecordSetting';
export * from './queryReplaySetting';
export * from './recordDynamicClass';
export * from './updateConfigTemplate';
export * from './updateInterfaceResponse';
export * from './updateRecordSetting';
export * from './updateReplaySetting';
