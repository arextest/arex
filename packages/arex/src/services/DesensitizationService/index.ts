export * from './deleteDesensitization';
export * from './listDesensitization';
export * from './saveDesensitization';
