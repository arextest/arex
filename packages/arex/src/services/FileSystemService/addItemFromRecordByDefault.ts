import { request } from '@/utils';

export type AddItemFromRecordByDefaultReq = {
  workspaceId: string;
  appName: string;
  interfaceName: string;
  nodeName: string;
  planId: string;
  operationId: string;
  recordId: string;
};

export async function dddItemFromRecordByDefault(params: AddItemFromRecordByDefaultReq) {
  return request
    .post<{ success: boolean }>('/report/filesystem/addItemFromRecordByDefault', params)
    .then((res) => res.body);
}
