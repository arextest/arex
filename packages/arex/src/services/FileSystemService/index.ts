export * from './collection';
export * from './queryDebuggingCase';
export * from './queryInterface';
export * from './request';
export * from './saveInterface';
export * from './workspace';
