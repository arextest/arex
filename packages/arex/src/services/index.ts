export * as ApplicationService from './ApplicationService';
export * as ComparisonService from './ComparisonService';
export * as ConfigService from './ConfigService';
export * as DesensitizationService from './DesensitizationService';
export * as EnvironmentService from './EnvironmentService';
export * as FileSystemService from './FileSystemService';
export * as LoginService from './LoginService';
export * as ReportService from './ReportService';
export * as ScheduleService from './ScheduleService';
export * as StorageService from './StorageService';
export * as UserService from './UserService';
