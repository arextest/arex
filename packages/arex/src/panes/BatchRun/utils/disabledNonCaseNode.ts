import { CollectionNodeType } from '@/constant';
import { CollectionType } from '@/services/FileSystemService';

/**
 * 禁用无 case 的非 case 节点
 * @param nodeData
 */
export default function disabledNonCaseNode(nodeData: CollectionType[]) {
  nodeData.forEach((item) => {
    item.disabled =
      item.nodeType !== CollectionNodeType.case && item.children?.length
        ? !disabledNonCaseNode(item.children).hasCase
        : item.nodeType !== CollectionNodeType.case;
  });
  return { nodeData, hasCase: nodeData.some((item) => !item.disabled) };
}
