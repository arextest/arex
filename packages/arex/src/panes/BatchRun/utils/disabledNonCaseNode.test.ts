import { describe, expect, test, vi } from 'vitest';

import { CollectionType } from '@/services/FileSystemService';

import disabledNonCaseNode from './disabledNonCaseNode';

vi.mock('@/constant', () => ({
  CollectionNodeType: {
    interface: 1,
    case: 2,
    folder: 3,
  },
}));

describe('disabledNonCaseNode', () => {
  test('no case collection tree types', () => {
    const collectionTreeData = [
      {
        key: '647849827137ad5ad372bd45',
        nodeName: 'node',
        nodeType: 3,
        infoId: '647849827137ad5ad372bd45',
        method: null,
        caseSourceType: 0,
        labelIds: null,
        children: [
          {
            key: '647849917137ad5ad372bd59',
            nodeName: 'New Request',
            nodeType: 1,
            infoId: '647849917137ad5ad372bd59',
            method: 'GET',
            caseSourceType: 0,
            labelIds: null,
            children: [],
          },
        ],
      },
    ] as CollectionType[];

    expect(disabledNonCaseNode(collectionTreeData)).toMatchSnapshot();
  });

  test('collection tree types with case', () => {
    const collectionTreeData = [
      {
        nodeName: 'node',
        nodeType: 3,
        key: '647849827137ad5ad372bd45',
        infoId: '647849827137ad5ad372bd45',
        method: null,
        caseSourceType: 0,
        labelIds: null,
        children: [
          {
            nodeName: 'New Request',
            nodeType: 1,
            key: '647849917137ad5ad372bd59',
            infoId: '647849917137ad5ad372bd59',
            method: 'GET',
            caseSourceType: 0,
            labelIds: null,
            children: [],
          },
        ],
      },
      {
        nodeName: 'batchcompare',
        nodeType: 3,
        key: '64536864a79d52663af78b33',
        infoId: '64536864a79d52663af78b33',
        method: null,
        caseSourceType: 0,
        labelIds: null,
        children: [
          {
            nodeName: 'retryBatchCompare',
            nodeType: 1,
            key: '64741a0d2a3299652af9111a',
            infoId: '648a7923bc7e1a5853cf6a33',
            method: 'POST',
            caseSourceType: 0,
            labelIds: null,
            children: [],
          },
          {
            nodeName: 'queryBatchCompareCaseMsgWithDiff',
            nodeType: 1,
            key: '647414772a3299652af90de4',
            infoId: '64741a0d2a3299652af9111a',
            method: 'POST',
            caseSourceType: 0,
            labelIds: null,
            children: [],
          },
          {
            nodeName: 'queryDiffInfoByCaseId',
            nodeType: 1,
            key: '647414772a3299652af90de4',
            infoId: '647414772a3299652af90de4',
            method: 'POST',
            caseSourceType: 0,
            labelIds: null,
            children: [
              {
                nodeName: '异常情况',
                nodeType: 2,
                key: '647434ee2a3299652af92445',
                infoId: '647434ee2a3299652af92445',
                method: null,
                caseSourceType: 2,
                labelIds: null,
                children: [],
              },
            ],
          },
          {
            nodeName: '/updateBatchCompareCase',
            nodeType: 1,
            key: '645374a7a79d52663af793ea',
            infoId: '645374a7a79d52663af793ea',
            method: 'POST',
            caseSourceType: 0,
            labelIds: null,
            children: [],
          },
          {
            nodeName: '/initBatchCompareReport',
            nodeType: 1,
            key: '645369aca79d52663af78bde',
            infoId: '645369aca79d52663af78bde',
            method: 'POST',
            caseSourceType: 0,
            labelIds: null,
            children: [
              {
                nodeName: 'case',
                nodeType: 2,
                key: '648a7792bc7e1a5853cf694c',
                infoId: '648a7792bc7e1a5853cf694c',
                method: null,
                caseSourceType: 2,
                labelIds: null,
                children: [],
              },
            ],
          },
          {
            nodeName: '/queryBatchCompareHistoryReport',
            nodeType: 1,
            key: '6453686ba79d52663af78b3b',
            infoId: '6453686ba79d52663af78b3b',
            method: 'POST',
            caseSourceType: 0,
            labelIds: null,
            children: [],
          },
        ],
      },
    ] as CollectionType[];

    expect(disabledNonCaseNode(collectionTreeData)).toMatchSnapshot();
  });
});
