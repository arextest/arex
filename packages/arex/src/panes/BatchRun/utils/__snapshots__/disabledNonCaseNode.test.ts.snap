// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`disabledNonCaseNode > collection tree types with case 1`] = `
{
  "hasCase": true,
  "nodeData": [
    {
      "caseSourceType": 0,
      "children": [
        {
          "caseSourceType": 0,
          "children": [],
          "disabled": true,
          "infoId": "647849917137ad5ad372bd59",
          "key": "647849917137ad5ad372bd59",
          "labelIds": null,
          "method": "GET",
          "nodeName": "New Request",
          "nodeType": 1,
        },
      ],
      "disabled": true,
      "infoId": "647849827137ad5ad372bd45",
      "key": "647849827137ad5ad372bd45",
      "labelIds": null,
      "method": null,
      "nodeName": "node",
      "nodeType": 3,
    },
    {
      "caseSourceType": 0,
      "children": [
        {
          "caseSourceType": 0,
          "children": [],
          "disabled": true,
          "infoId": "648a7923bc7e1a5853cf6a33",
          "key": "64741a0d2a3299652af9111a",
          "labelIds": null,
          "method": "POST",
          "nodeName": "retryBatchCompare",
          "nodeType": 1,
        },
        {
          "caseSourceType": 0,
          "children": [],
          "disabled": true,
          "infoId": "64741a0d2a3299652af9111a",
          "key": "647414772a3299652af90de4",
          "labelIds": null,
          "method": "POST",
          "nodeName": "queryBatchCompareCaseMsgWithDiff",
          "nodeType": 1,
        },
        {
          "caseSourceType": 0,
          "children": [
            {
              "caseSourceType": 2,
              "children": [],
              "disabled": false,
              "infoId": "647434ee2a3299652af92445",
              "key": "647434ee2a3299652af92445",
              "labelIds": null,
              "method": null,
              "nodeName": "异常情况",
              "nodeType": 2,
            },
          ],
          "disabled": false,
          "infoId": "647414772a3299652af90de4",
          "key": "647414772a3299652af90de4",
          "labelIds": null,
          "method": "POST",
          "nodeName": "queryDiffInfoByCaseId",
          "nodeType": 1,
        },
        {
          "caseSourceType": 0,
          "children": [],
          "disabled": true,
          "infoId": "645374a7a79d52663af793ea",
          "key": "645374a7a79d52663af793ea",
          "labelIds": null,
          "method": "POST",
          "nodeName": "/updateBatchCompareCase",
          "nodeType": 1,
        },
        {
          "caseSourceType": 0,
          "children": [
            {
              "caseSourceType": 2,
              "children": [],
              "disabled": false,
              "infoId": "648a7792bc7e1a5853cf694c",
              "key": "648a7792bc7e1a5853cf694c",
              "labelIds": null,
              "method": null,
              "nodeName": "case",
              "nodeType": 2,
            },
          ],
          "disabled": false,
          "infoId": "645369aca79d52663af78bde",
          "key": "645369aca79d52663af78bde",
          "labelIds": null,
          "method": "POST",
          "nodeName": "/initBatchCompareReport",
          "nodeType": 1,
        },
        {
          "caseSourceType": 0,
          "children": [],
          "disabled": true,
          "infoId": "6453686ba79d52663af78b3b",
          "key": "6453686ba79d52663af78b3b",
          "labelIds": null,
          "method": "POST",
          "nodeName": "/queryBatchCompareHistoryReport",
          "nodeType": 1,
        },
      ],
      "disabled": false,
      "infoId": "64536864a79d52663af78b33",
      "key": "64536864a79d52663af78b33",
      "labelIds": null,
      "method": null,
      "nodeName": "batchcompare",
      "nodeType": 3,
    },
  ],
}
`;

exports[`disabledNonCaseNode > no case collection tree types 1`] = `
{
  "hasCase": false,
  "nodeData": [
    {
      "caseSourceType": 0,
      "children": [
        {
          "caseSourceType": 0,
          "children": [],
          "disabled": true,
          "infoId": "647849917137ad5ad372bd59",
          "key": "647849917137ad5ad372bd59",
          "labelIds": null,
          "method": "GET",
          "nodeName": "New Request",
          "nodeType": 1,
        },
      ],
      "disabled": true,
      "infoId": "647849827137ad5ad372bd45",
      "key": "647849827137ad5ad372bd45",
      "labelIds": null,
      "method": null,
      "nodeName": "node",
      "nodeType": 3,
    },
  ],
}
`;
